src/components/dashboard/data/utils.js

```diff
@@ -1,30 +1,176 @@
+    LEARNER_ACKNOWLEDGED_ASSIGNMENT_CANCELLATION_ALERT,
+  );
+  const isCanceled = assignment.state === ASSIGNMENT_TYPES.CANCELED;
+  const hasDismissedCancelation = assignment.actions.some((action) => {
+    const isCanceledNoticationAction = [
+      ASSIGNMENT_ACTION_TYPES.CANCELED,
+      ASSIGNMENT_ACTION_TYPES.AUTOMATIC_CANCELATION,
+    ].includes(action.actionType);
+    const isAcknowledged = dayjs(action.completedAt).isBefore(new Date(lastCanceledAlertDismissedTime));
+    return isCanceled && isCanceledNoticationAction && isAcknowledged;
+  });
+  return {
+    isCanceled,
+    hasDismissedCancelation,
+  };
+}
+
+/**
+ * Determines whether there are any unacknowledged canceled assignments.
+ *
+ * @param {Array} assignments - Metadata about the assignments.
+ * @returns {Boolean} - Returns true if there are any unacknowledged canceled assignments, otherwise false.
+ */
+export function getHasUnacknowledgedCanceledAssignments(assignments) {
+  return assignments.some((assignment) => {
+    const { isCanceled, hasDismissedCancelation } = isCanceledAssignmentAcknowledged(assignment);
+    return isCanceled && !hasDismissedCancelation;
+  });
+}
+
+/**
+ * Takes a flattened array of assignments and returns an object containing
+ * lists of assignments for each assignment state.
  *
  * @param {Array} assignments - List of content assignments.
  * @returns {{
  *  assignments: Array,
  *  hasAssignments: Boolean,
- *  hasActiveAssignments: Boolean,
- *  activeAssignments: Array,
+ *  allocatedAssignments: Array,
+ *  hasAllocatedAssignments: Boolean,
+ *  canceledAssignments: Array,
+ *  hasCanceledAssignments: Boolean,
+ *  acceptedAssignments: Array,
+ *  hasAcceptedAssignments: Boolean,
  * }}
  */
-export default function getActiveAssignments(assignments = []) {
-  const activeAssignments = assignments.filter((assignment) => [
-    ASSIGNMENT_TYPES.CANCELED, ASSIGNMENT_TYPES.ALLOCATED,
-  ].includes(assignment.state));
-  const hasAssignments = assignments.length > 0;
-  const hasActiveAssignments = activeAssignments.length > 0;
+export function getAssignmentsByState(assignments = []) {
+  const allAssignments = [];
+  const allocatedAssignments = [];
+  const canceledAssignments = [];
+  const acceptedAssignments = [];
+
+  assignments.forEach((assignment) => {
+    allAssignments.push(assignment);
+    if (assignment.state === ASSIGNMENT_TYPES.ALLOCATED) {
+      allocatedAssignments.push(assignment);
+    }
+    if (assignment.state === ASSIGNMENT_TYPES.CANCELED) {
+      canceledAssignments.push(assignment);
+    }
+    if (assignment.state === ASSIGNMENT_TYPES.ACCEPTED) {
+      acceptedAssignments.push(assignment);
+    }
+  });
+
+  const hasAssignments = allAssignments.length > 0;
+  const hasAllocatedAssignments = allocatedAssignments.length > 0;
+  const hasCanceledAssignments = canceledAssignments.length > 0;
+  const hasAcceptedAssignments = acceptedAssignments.length > 0;
```

Comment:
[suggestion/question] I am wondering if it'd be worth capturing the `ASSIGNMENT_TYPS.ERRORED` assignments as well to flesh out the utility of this function

Reply:
Couldn't hurt. Not currently included because nothing in this repo refers to errored assignments. If/when it does though, having this hook already account for errored assignments would be helpful. Will do!